homeassistant/components/media_player/onkyo.py

```diff
@@ -47,9 +46,40 @@
         vol.All(vol.Coerce(int), vol.Range(min=1, max=SUPPORTED_MAX_VOLUME)),
     vol.Optional(CONF_SOURCES, default=DEFAULT_SOURCES):
         {cv.string: cv.string},
-    vol.Optional(CONF_ZONE2, default=False): cv.boolean,
 })
 
+TIMEOUT_MESSAGE = 'Timeout waiting for response.'
+
+
+def determine_zones(host):
+    """Determine what zones are available for the receiver"""
+    import eiscp
+    out = {
+        "zone2": False,
+        "zone3": False,
+    }
+    receiver = eiscp.eISCP(host)
```

Comment:
Any reason we would make a new connection and not re-use the existing connection ?

Reply:
Nope, will fix.